contracts/Auction.sol

```diff
@@ -139,6 +141,14 @@ contract Auction is IAuction {
         return self.amountOutstanding;
     }
 
+    function amountTransferred() external view returns (uint256) {
+        // Need to check whether the auction is open as harikari is done
+        // upon auction close so there is no possibility to determine
+        // the transferred amount on a closed auction.
+        require(self.amountOutstanding > 0, "Auction must be open");
```

Comment:
I am not sure if I follow. `selfdestruct` should destroy the contract so that it cannot be called in the future. This is only a problem in the *current* transaction and if I understand correctly, we want to make sure functions are called in the right order in `Auctioneer.earlyClose` or in the code calling `Auctioneer.earlyClose`, so that the amount transferred is evaluated before `Auction.earlyClose`.

If so, why don't we return `surplus = amountInitial - amountOutstanding` from `earlyClose`? We do not need this function then, the risk is gone and we consume less gas. This would also play well with another comment I left in `RiskMangerV1` about having the surplus returned from `Auctioneer.earlyClose`. WDYT?

Reply:
Done in https://github.com/keep-network/coverage-pools/pull/43/commits/ba3de981619c6aea018188c28b64fc2e99f4ce49

Reply:
Right, I missed this fact. Anyway, I'd remove the `require` check from here. The `Auctioneer` needs to fetch all the information it needs before calling `earlyClose()`. This is true for `amountTransferred` but also for any other field or function in `Auction` contract: `amountInitial`, `amountOutstanding`, `startTime`... I do not think we should treat `amountTransferred()` in any special way.

Reply:
https://github.com/keep-network/coverage-pools/pull/43/commits/495e090974a40c594c8304435a477d3c5eab2fe6